apiserver/facades/client/application/deploy.go

```diff
@@ -142,19 +143,19 @@ func getEffectiveBindingsForCharmMeta(charmMeta *charm.Meta, givenBindings map[s
 	}
 
 	// Get the application-level default binding for all unspecified endpoints, if
-	// set. Otherwise use the empty default.
-	applicationDefaultSpace, defaultSupplied := givenBindings[""]
-	if defaultSupplied {
-		// Record that a default binding was requested
-		defaultBindings[""] = applicationDefaultSpace
+	// set. Otherwise use the DefaultSpaceId.
+	applicationDefaultSpaceID, defaultSupplied := givenBindings[""]
+	if !defaultSupplied {
+		applicationDefaultSpaceID = network.DefaultSpaceId
 	}
+	defaultBindings[""] = applicationDefaultSpaceID
```

Comment:
Ideally we should be using the (currently not exported) merge bindings code from state to calculate the effective bindings